core/shared/src/main/scala/org/http4s/headers/Cookie.scala

```diff
@@ -31,19 +31,9 @@ object Cookie {
   def parse(s: String): ParseResult[Cookie] =
     ParseResult.fromParser(parser, "Invalid Cookie header")(s)
 
-  private[http4s] val parser: Parser[Cookie] = {
-    import Parser.{char, string}
-
-    /* cookie-string = cookie-pair *( ";" SP cookie-pair ) */
-    val cookieString = (RequestCookie.parser ~ (string("; ") *> RequestCookie.parser).rep0).map {
-      case (head, tail) =>
-        Cookie(NonEmptyList(head, tail))
-    }
-
-    /* We also see trailing semi-colons in the wild, and grudgingly tolerate them
-     * here. */
-    cookieString <* char(';').?
```

Comment:
The fact that there is already grudging tolerance for trailing semi-colons suggests there is already some precedence to be more lenient than the specification.

Personally I don't think I have enough experience to know when to make these sorts of exceptions. A more experienced maintainer will have to weigh in.